pe/models.py

```diff
@@ -17,28 +26,59 @@ class Exam(models.Model):
     report = models.ForeignKey(to='Report', related_name='exams', null=True, on_delete=models.SET_NULL)
     course_id = models.IntegerField(verbose_name='Canvas Course ID for Exam')
     assignment_id = models.IntegerField(verbose_name='Canvas Assignment ID for Exam', unique=True)
+    default_time_filter = models.DateTimeField(verbose_name='Earliest Date & Time for Submission Search')
 
     def __str__(self):
         return (
             f'(id={self.id}, sa_code={self.sa_code}, name={self.name}, report={self.report}, ' +
-            f'course_id={self.course_id}, assignment_id={self.assignment_id})'
+            f'course_id={self.course_id}, assignment_id={self.assignment_id}, ' +
+            f'default_time_filter={self.default_time_filter})'
         )
 
+    def get_last_sub_graded_datetime(self) -> Union[datetime, None]:
+        """
+        Return latest graded_timestamp value for an exam's submissions, or None.
+
+        :return: Either a datetime object or null.
+        :rtype: datetime.datetime or None
+        """
+        last_graded_dt = None
+        sub_ordered_queryset = self.submissions.order_by('-graded_timestamp')
```

Comment:
`exam.submissions.order_by('-graded_timestamp') should return the graded timestamps in descending order, which then gives us the last one, right?

In terms of whether there is a need for this, this is how I determine where to start pulling submissions from for Canvas (i.e. the `graded_since` value in the Canvas parameters). Does that make sense?

Reply:
Since the ScoresOrchestration process will be executed separately for different exams, I think a filter by exam code may be necessary, like exam.submissions.filter(exam=<exam_id>).order_by('-graded_timestamp'), so that there is no time gap for every exam types.

Reply:
Sam demonstrated the Exam model contains the exam id, so having the order_by is enough for the graded_timestamp selection.